### PR TITLE
Changed propType's to object for `component` prop

### PR DESCRIPTION
**Pull request by guschnwg**
Fixes `component` prop of <FormField /> component.

Fixes: https://github.com/grommet/grommet/issues/2981

### CODE REVIEW

**Comment by oorestisime** [comment]
Hey thanks for jumping on this. do you think we need another story on this? i am thinking you could modify the existing one.

**Comment by guschnwg** [comment]
I think it is not needed after all.

My tought was "well, i can do both ways `htmlFor="text-area"` and `component={TextArea}`" but since all other examples as using `htmlFor` way, there is no need to another case.

**Comment by oorestisime** [comment]
yeah i think we can change the existing textarea example to llustrate that way of doing it but adding an extra story would make it quite verbose!

File: src/js/components/FormField/formfield.stories.js [hunk 1]
```diff
@@ -55,9 +55,7 @@ class FormFieldTextInput extends Component {
 const FormFieldTextArea = props => (
   <Grommet theme={grommet}>
     <Box align="center" pad="large">
-      <FormField label="Label" htmlFor="text-area" {...props}>
-        <TextArea id="text-area" placeholder="placeholder" />
-      </FormField>
+      <FormField label="Label" component={TextArea} {...props} />
```
[the remark this answers is not in the document]
Comment (oorestisime) [reply src/js/components/FormField/formfield.stories.js]: by transparent you mean?

[the remark this answers is not in the document]
Comment (ShimiSun) [reply src/js/components/FormField/formfield.stories.js]: ...put back the placeholder as it was. so the user won't notice it was changed behind the scene, so the change will be transparent, sorry if it wasn't clear.

**Comment by mrebb** [comment]
@guschnwg Could you test this PR for Form story and provide a fix?  Storybook example for `Form` broke with this PR.

**Comment by guschnwg** [comment]
@mrebb my mistake, i forgot to undo that

**Pull request merged**